File: src/main/java/net/thedragonteam/thedragonlib/client/util/ClientUtills.java
package net.thedragonteam.thedragonlib.client.util;

import net.thedragonteam.thedragonlib.util.LogHelper;

import java.net.URI;

public class ClientUtills {

    public static void openLink(String url) {
        try {
            URI uri = new URI(url);
            Class oclass = Class.forName("java.awt.Desktop");
            Object object = oclass.getMethod("getDesktop", new Class[0]).invoke(null);
            oclass.getMethod("browse", new Class[]{URI.class}).invoke(object, uri);
        } catch (Throwable throwable) {
            LogHelper.error("Couldn\'t open link");
            throwable.printStackTrace();
        }
    }
}

File: src/main/java/net/thedragonteam/thedragonlib/client/util/ClientUtills.kt
package net.thedragonteam.thedragonlib.client.util

import net.thedragonteam.thedragonlib.util.LogHelper

import java.net.URI

object ClientUtills {

    fun openLink(url: String) {
        try {
            val uri = URI(url)
            val oclass = Class.forName("java.awt.Desktop")
            val `object` = oclass.getMethod("getDesktop", *arrayOfNulls<Class<*>>(0)).invoke(null)
            oclass.getMethod("browse", *arrayOf<Class<*>>(URI::class.java)).invoke(`object`, uri)
        } catch (throwable: Throwable) {
            LogHelper.error("Couldn\'t open link")
            throwable.printStackTrace()
        }

    }
}